client/src/main/scala/org/http4s/client/middleware/Retry.scala

```diff
@@ -19,7 +20,7 @@ object Retry {
 
   private[this] val logger = getLogger
 
-  def apply[F[_]](policy: RetryPolicy[F])(client: Client[F])(
+  def apply[F[_]](policy: RetryPolicy[F])(client: Client[F])(redactHeaderWhen: CaseInsensitiveString => Boolean = Headers.SensitiveHeaders.contains)(
```

Comment:
I'd make this the second arg in the first list.  It's nice for middleware composition when the `client` arg stands alone in the last list.